src/back-end/lib/resources/proposal/team-with-us.ts

```diff
@@ -729,12 +742,17 @@ const update: crud.Update<
               ),
               await validateTWUProposalTeam(
                 connection,
-                validatedTWUProposal.value.team,
-                request.id
+                validatedTWUProposal.value.team.map(
+                  ({ member, hourlyRate }) => ({
+                    member: member.id,
+                    hourlyRate
+                  })
+                ),
```

Comment:
I wonder if we'd be better off changing the type of the team in `update`? It differs from that of `create`.

Reply:
Not sure what you mean?

Reply:
eh, kinda hard to explain. i need to do this mapping because of
```
CreateRequestBody {
  opportunity: Id;
  organization?: Id;
  attachments: Id[];
  resourceQuestionResponses: CreateTWUProposalResourceQuestionResponseBody[];
  status: CreateTWUProposalStatus;
  team: CreateTWUTeamMemberBody[];
}
```
but the type of a a stored team member is
```
export interface TWUProposalTeamMember {
  member: UserSlim;
  pending: boolean;
  idpUsername: string;
  hourlyRate: number;
}
```
I think the mapping is fine, but it was difficult to debug because `validateTWUProposalTeam`'s raw argument has type `any`, so I'd like to avoid that headache if the situation comes again (though it might not?)

Reply:
Oh, I see what you mean. Since the difficulty with debugging is caused by the use of `any[]` in `validateTWUProposalTeam`, it might make sense to update that function signature to have a `raw: { member: string, hourlyRate: number }[]` type instead. This would force you to run `getString` and `getNumber` at the call site (as needed), but might result in a more sustainable solution with better type errors.